src/router.rs

```diff
@@ -13,15 +13,17 @@ struct MatcherKey(usize, Uuid);
 pub struct Router<'a> {
     schema: &'a Schema,
     matchers: BTreeMap<MatcherKey, Expression>,
-    pub fields: HashMap<String, usize>,
+    pub fields: Vec<(String, usize)>, // fileds array of tuple(name, count)
+    pub fields_map: HashMap<String, usize>, // field name -> index map
```

Comment:
`fields` is an flat array to store all field names and corresponding counts.
Here we uses an hashmap `fields_map` to store the field's index in `fields` array, `fields_map` is not used in runtime, but only for router matcher add and delete, when matcher changed, use this map to quickly get the index of fields.